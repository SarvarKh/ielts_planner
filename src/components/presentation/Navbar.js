import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';

const Navbar = ({ title }) => {
  const history = useHistory();

  const logOut = () => {
    sessionStorage.removeItem('token');
    history.push('/log_in');
  };
  return (
    <nav>
      <div className="n-title">{title}</div>
      <button type="button" onClick={logOut}>
        <i className="fas fa-power-off" />
      </button>
    </nav>
  );
};

export default Navbar;

Navbar.propTypes = {
  title: PropTypes.string.isRequired,
};
