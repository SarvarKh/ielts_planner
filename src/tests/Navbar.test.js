import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from '../components/presentation/Navbar';

it('renders correctly', () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
