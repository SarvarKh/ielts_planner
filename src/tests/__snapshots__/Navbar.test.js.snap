// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav>
  <div
    className="n-title"
  />
  <button
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-power-off"
    />
  </button>
</nav>
`;
